code/datums/elements/blocks_explosives.dm

```diff
@@ -20,6 +20,13 @@
 
 /datum/element/blocks_explosives/Detach(datum/source)
 	. = ..()
+	if(ismovable(source))
```

Comment:
this is unneeded, source cannot be anything but atom/movable here, so you can just drop this typecheck and the cast you do on the next line

Reply:
Can I do the cast on the proc parameter without causing problems? I think the casting would be useful for code editors.

Reply:
yep, that's fine byond won't care and only atoms should get there regardless